src/ainftBase.ts

```diff
@@ -0,0 +1,31 @@
+import Ain from "@ainblockchain/ain-js";
+import { AINFT_SERVER_ENDPOINT } from "./constants";
+import stringify = require("fast-json-stable-stringify");
+
+export default class AinftBase {
+  public baseUrl: string;
+  public route: string;
+  public ain: Ain;
+
+  constructor(
+    baseUrl = AINFT_SERVER_ENDPOINT,
+    ain: Ain,
+    route?: string,
+  ) {
+    this.route = route || '';
+    this.baseUrl = baseUrl + this.route;
```

Comment:
minor) what about using setBaseUrl()?

Reply:
If a class member is not initialized in define and in the constructor, an error will occur and so we will not be able to use setBaseUrl in the constructor.

Reply:
question) do you mean it's true in ts in general? or just in this case?
i'm asking this because we already have similar code in js:
https://github.com/ainblockchain/ain-blockchain/blob/master/db/state-manager.js#L15 

i think repeated code is okay if it's just simple (e.g. assignment) but if it includes some logic, i think we can try to avoid duplication if possible?

Reply:
In our case, we defined base url as `private baseUrl: string` before the constructor. in class, if class member is not initialized when define and in the constructor. it occurred error.

But In the case you referenced, `rootMap` class member is not defined before the constructor. so It did not occur error.

If we do not define `baseUrl` before the constructor, we can use `setBaseUrl`.  But I think defining class members before constructor is more helpful to read code.

Reply:
oh, i see. thanks for explanation.
then how about initializing with a primitive value in the definition? e.g.
```
export default class AinftBase {
  public baseUrl: string = '';
  ...

  constructor(
    ...
    baseUrl = AINFT_SERVER_ENDPOINT,
    ...
  ) {
    ...
    this.setBaseUrl(baseUrl);
    ...
  }
```
or
```
export default class AinftBase {
  public baseUrl: string = AINFT_SERVER_ENDPOINT;
  ...

  constructor(
    ...
    baseUrl?: string,
    ...
  ) {
    ...
    if (baseUrl) {
      this.setBaseUrl(baseUrl);
    }
    ...
  }
```

Reply:
Oh, the second suggestion looks good! I will apply it in the next PR. Thanks!!

Reply:
thanks! 👌